src/hssm/likelihoods/analytical.py

```diff
@@ -262,15 +241,17 @@ def logp_ddm(
     z_flipped = pt.switch(flip, 1 - z, z)  # transform z if x is upper-bound response
     rt = rt - t
 
-    p = pt.maximum(ftt01w(rt, a, z_flipped, err, k_terms), pt.exp(LOGP_LB))
+    negative_rt = rt <= epsilon
 
-    logp = pt.where(
-        rt <= epsilon,
-        LOGP_LB,
+    tt = negative_rt * epsilon + (1 - negative_rt) * rt
+
+    p = pt.maximum(ftt01w(tt, a, z_flipped, err, k_terms), pt.exp(LOGP_LB))
+
+    logp = negative_rt * LOGP_LB + (1 - negative_rt) * (
         pt.log(p)
         - v_flipped * a * z_flipped
-        - (v_flipped**2 * rt / 2.0)
-        - 2.0 * pt.log(a),
+        - (v_flipped**2 * tt / 2.0)
+        - 2.0 * pt.log(pt.maximum(epsilon, a))
     )
 
     checked_logp = check_parameters(logp, a >= 0, msg="a >= 0")
```

Comment:
in the spirit of above, this check could be `a>0` but honestly we shouldn't really ever get there.

Reply:
Same as above